server-common/src/main/java/org/apache/kafka/server/common/MetadataVersion.java

```diff
@@ -344,7 +348,9 @@ public boolean isControllerRegistrationSupported() {
     }
 
     public short partitionChangeRecordVersion() {
-        if (isElrSupported()) {
+        if (isDirectoryAssignmentSupported()) {
+            return (short) 2;
+        } else if (isElrSupported()) {
```

Comment:
We are going to need to do the same thing in `partitionRecordVersion()` and `registerBrokerRecordVersion()`.  There is no `brokerRegistrationChangeRecordVersion()`, and it appears we don't need that yet since we explicitly send either version 0 or version 1.